controllers/clusterversion_controller.go

```diff
@@ -127,12 +129,25 @@ func (c *ClusterVersionReconciler) SetupWithManager(mgr ctrl.Manager) error {
 		),
 	)
 
+	storageClientPredicate := builder.WithPredicates(
+		predicate.NewPredicateFuncs(
+			func(obj client.Object) bool {
+				annotations := obj.GetAnnotations()
+				if desiredChannel, ok := annotations[utils.DesiredSubscriptionChannelAnnotationKey]; ok {
+					return c.subscriptionChannel != desiredChannel
+				}
+				return false
+			},
+		),
+	)
```

Comment:
Having a predicate based on an internal (not external) reconciler state will not work the way you expect. It will filter events as they queue with unpredictable results. Why do we need to filter reconciliation on clients?

Reply:
sry, I was also a bit skeptical to use it. My intention is to trigger a reconcile when the existing subscription channel is different compared to storageclient desired channel annotation.

I don't want to send reconciles every minute as job updates the annotations value of storageclient, what I could do is queue an event if annotation value changes which isn't fool proof though.

Reply:
This seems like a bug in the job. The actual API update call should only happen if the annotation has changed
We have a util that sets the annotation and returns a boolean to indicate if a change actually happened (in memory) and in most cases, we call `client.Update` only if that return value is true.

Reply:
done.